source/host_resolver.c

```diff
@@ -723,3 +723,95 @@ int aws_host_resolver_init_default(
 
     return AWS_OP_SUCCESS;
 }
+
+#ifdef AWS_USE_LIBUV
+
+#    include <uv.h>
+
+struct uv_host_resolver {
+    /* This needs to be the first element in the struct so the pointers are safely castable */
+    struct default_host_resolver impl;
+    uv_loop_t *uv_loop;
+};
+
+struct uv_resolve_request {
+    aws_on_host_resolved_result_fn *resolved_cb;
+    void *resolved_ud;
+    /* Handle to increase uv_loop's ref count */
+    uv_async_t async;
+};
+
+static void uv_host_resolved(
+    struct aws_host_resolver *resolver,
+    const struct aws_string *host_name,
+    int err_code,
+    const struct aws_array_list *host_addresses,
+    void *user_data) {
+
+    struct uv_resolve_request *request = user_data;
+
+    request->resolved_cb(resolver, host_name, err_code, host_addresses, request->resolved_ud);
+
+    uv_close((uv_handle_t *)&request->async, NULL);
+}
+
+static int uv_resolve_host(
+    struct aws_host_resolver *resolver,
+    const struct aws_string *host_name,
+    aws_on_host_resolved_result_fn *res,
+    struct aws_host_resolution_config *config,
+    void *user_data) {
+
+    struct uv_host_resolver *uv_host_resolver = resolver->impl;
+
+    struct uv_resolve_request *request = aws_mem_acquire(resolver->allocator, sizeof(struct uv_resolve_request));
+    request->resolved_cb = res;
+    request->resolved_ud = user_data;
+    uv_async_init(uv_host_resolver->uv_loop, &request->async, NULL);
+
+    return default_resolve_host(resolver, host_name, uv_host_resolved, config, request);
+}
+
+static struct aws_host_resolver_vtable s_uv_vtable = {
+    .purge_cache = resolver_purge_cache,
+    .resolve_host = uv_resolve_host,
+    .record_connection_failure = resolver_record_connection_failure,
+    .destroy = resolver_destroy,
```

Comment:
Is this actually going to work? resolver_destroy doesn't delete the entire resolver, it only deletes the inner default resolver.

Reply:
OH NO good catch

Reply:
ACTUALLY yeah it should work, because the only extra thing on uv_host_resolver is the uv_loop ptr, and that doesn't need any cleanup.